Generate.py

```diff
@@ -200,9 +200,11 @@ def main(args=None) -> Tuple[argparse.Namespace, int]:
                             except Exception as e:
                                 raise Exception(f"Error setting {k} to {v} for player {player}") from e
 
+                    if not getattr(erargs, "name", None):
+                        setattr(erargs, "name", {})  # if the first slot does not define a name we won't have a name obj to look up
```

Comment:
Somewhat of a nitpick, but I feel like there should just be a `setattr(erargs, "name", {})` before the while loop

Reply:
iirc if any yaml has a real name the dict will be made already, so doing that will override existing names
but if none do the attr doesn't exist so the lookups fail, hence the `if`
if you want me to just move the two lines i have no issues though

Reply:
confirmed that this is the case, so unless you just want the names dict blindly initialized in the erargs namespace earlier this is not a simple fix

Reply:
Yeah, the creation of erargs happens after this line:

```
    if args.multi == 0:
        raise ValueError(
            "No individual player files found and number of players is 0. "
            "Provide individual player files or specify the number of players via host.yaml or --multi."
        )
```

Which to me indicates that it would be safe to just `setattr(erargs, "names", {})` right after its creation. What do you think about that?

Reply:
no qualms, will do some testing around that later